server/backend/neighbors.py

```diff
@@ -23,6 +23,15 @@
 # (pre-populated on server startup if .config.WARM_CACHE is true)
 word_models = None
 
+# Conccept ID Mapper
+concept_id_mapper = pd.read_csv("all_concept_ids.tsv.xz", sep="\t") >> ply.define(
```

Comment:
>Also, just for what it's worth, I'm not a big fan of expensive operations occurring just because you imported a module. I'd prefer to see this loading code wrapped in a function and called when needed, with the global variable set to a sentinel value (e.g., None) if it hasn't been called yet.

Ah okay. That's fair I can put it in the function. Just know that since its 'xz' compression it will take a minute or two to load. Hence my thought of spending that time on server load rather than each query.

Reply:
Agreed, we definitely wouldn't want it happening every query. What I'm proposing is similar to a singleton, i.e.:

```python
concept_id_mapper_dict = None

def get_concept_id_mapper():
    global concept_id_mapper_dict
  
    if not concept_id_mapper_dict:
        concept_id_mapper = (
            pd.read_csv(data_folder / Path("all_concept_ids.tsv.xz"), sep="\t")
            >> ply.define(concept_id="concept_id.str.lower()")
        )

        concept_id_mapper_dict = dict(
            zip(concept_id_mapper.concept_id.tolist(), concept_id_mapper.concept.tolist())
        )
    
    return concept_id_mapper_dict
```

When you need to use it, you'd call `get_concept_id_mapper()`, or just ensure that it's been called before you reference the global variable. If it's already been loaded it just gets returned. Since we want this to occur on server load, in `main.py` I'd also put the following:

```python
@app.on_event("startup")
def init_concept_map():
    # prepopulates concept_id_mapper_dict before anything requests it 
    get_concept_id_mapper()
```